qutebrowser/browser/history.py

```diff
@@ -452,6 +452,63 @@ def history_clear(force=False):
                               title="Clear all browsing history?")
 
 
+def _normalize_url(url: str) -> QUrl:
+    if not url:
+        return QUrl()
+    if '://' not in url:
+        url = f'https://{url}'
+    return QUrl(url)
+
+
+def _find_matching_history_urls(qurl: QUrl) -> list[str]:
+    if not qurl.isValid():
+        return []
+
+    target_host = qurl.host().lower()
+    matching_urls = []
+
+    for entry in web_history:
```

Comment:
This hangs qutebrowser for 7 seconds on my (modern) laptop with my history. I don't think manually iterating over history entries is the right approach here. If you only care about matching hostnames, this should be done using sqlite instead, with something suitable added to the `WebHistory` class to match and delete matching rows there.

This will be a bit trickier to implement properly (as e.g. `url_cleared` still needs to be emitted, and `History` / `CompletionHistory` need to still be kept in sync), but a multi-second UI hang is not acceptable unfortunately.

Reply:
thats a great callout, i was testing on a fresh install. ill think of something better